caldav/match.go

```diff
@@ -0,0 +1,224 @@
+package caldav
+
+import (
+	"time"
+
+	"github.com/emersion/go-ical"
+)
+
+// Filter returns the filtered list of calendar objects matching the provided query.
+// A nil query will return the full list of calendar objects.
+func Filter(query *CalendarQuery, cos []CalendarObject) ([]CalendarObject, error) {
+	if query == nil {
+		// FIXME: should we always return a copy of the provided slice?
+		return cos, nil
+	}
+
+	out := make([]CalendarObject, 0)
```

Comment:
Style nit: either specify the capacity (e.g. `len(cos)`), either switch to `var out []CalendarObject`.